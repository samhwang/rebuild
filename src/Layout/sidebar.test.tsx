import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import { BrowserRouter } from 'react-router-dom';
import { ColorScheme, ColorSchemeProvider } from '@mantine/core';
import { useToggle } from '@mantine/hooks';
import { ReactNode } from 'react';
import Sidebar from './sidebar';

function ColorSchemeShell({ children }: { children: ReactNode }) {
  const [colorScheme, toggleColorScheme] = useToggle<ColorScheme>([
    'light',
    'dark',
  ]);
  return (
    <ColorSchemeProvider
      colorScheme={colorScheme}
      toggleColorScheme={toggleColorScheme}
    >
      {children}
    </ColorSchemeProvider>
  );
}

describe('Sidebar', () => {
  it('Should render without crashing and match snapshot', () => {
    const tree = render(
      <BrowserRouter>
        <ColorSchemeShell>
          <Sidebar hidden={false} />
        </ColorSchemeShell>
      </BrowserRouter>
    );
    expect(tree).toMatchSnapshot();
  });

  it('Should be hidden if passed with hidden true', () => {
    const tree = render(
      <BrowserRouter>
        <ColorSchemeShell>
          <Sidebar hidden />
        </ColorSchemeShell>
      </BrowserRouter>
    );
    expect(tree).toMatchSnapshot();
  });
});
