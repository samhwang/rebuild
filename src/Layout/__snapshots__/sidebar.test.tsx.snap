// Vitest Snapshot v1

exports[`Sidebar > Should be hidden if passed with hidden true 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="mantine-Navbar-root mantine-iben9"
        data-hidden="true"
      >
        <div
          class="mantine-khtkeg"
        >
          <div
            class="mantine-1avyp1d"
          >
            <div
              class="mantine-Group-root mantine-19jxmdp"
            >
              <button
                class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-Group-child mantine-817y8t"
                type="button"
              >
                <svg
                  class="icon icon-tabler icon-tabler-moon-stars"
                  fill="none"
                  height="16"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                    stroke="none"
                  />
                  <path
                    d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z"
                  />
                  <path
                    d="M17 4a2 2 0 0 0 2 2a2 2 0 0 0 -2 2a2 2 0 0 0 -2 -2a2 2 0 0 0 2 -2"
                  />
                  <path
                    d="M19 11h2m-1 -1v2"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div
          class="mantine-Text-root mantine-1pbxw0k"
        >
          Sidebar Text
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="mantine-Navbar-root mantine-iben9"
      data-hidden="true"
    >
      <div
        class="mantine-khtkeg"
      >
        <div
          class="mantine-1avyp1d"
        >
          <div
            class="mantine-Group-root mantine-19jxmdp"
          >
            <button
              class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-Group-child mantine-817y8t"
              type="button"
            >
              <svg
                class="icon icon-tabler icon-tabler-moon-stars"
                fill="none"
                height="16"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                  stroke="none"
                />
                <path
                  d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z"
                />
                <path
                  d="M17 4a2 2 0 0 0 2 2a2 2 0 0 0 -2 2a2 2 0 0 0 -2 -2a2 2 0 0 0 2 -2"
                />
                <path
                  d="M19 11h2m-1 -1v2"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="mantine-Text-root mantine-1pbxw0k"
      >
        Sidebar Text
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Sidebar > Should render without crashing and match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="mantine-Navbar-root mantine-iben9"
      >
        <div
          class="mantine-khtkeg"
        >
          <div
            class="mantine-1avyp1d"
          >
            <div
              class="mantine-Group-root mantine-19jxmdp"
            >
              <button
                class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-Group-child mantine-817y8t"
                type="button"
              >
                <svg
                  class="icon icon-tabler icon-tabler-moon-stars"
                  fill="none"
                  height="16"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                    stroke="none"
                  />
                  <path
                    d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z"
                  />
                  <path
                    d="M17 4a2 2 0 0 0 2 2a2 2 0 0 0 -2 2a2 2 0 0 0 -2 -2a2 2 0 0 0 2 -2"
                  />
                  <path
                    d="M19 11h2m-1 -1v2"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div
          class="mantine-Text-root mantine-1pbxw0k"
        >
          Sidebar Text
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="mantine-Navbar-root mantine-iben9"
    >
      <div
        class="mantine-khtkeg"
      >
        <div
          class="mantine-1avyp1d"
        >
          <div
            class="mantine-Group-root mantine-19jxmdp"
          >
            <button
              class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-Group-child mantine-817y8t"
              type="button"
            >
              <svg
                class="icon icon-tabler icon-tabler-moon-stars"
                fill="none"
                height="16"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                  stroke="none"
                />
                <path
                  d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z"
                />
                <path
                  d="M17 4a2 2 0 0 0 2 2a2 2 0 0 0 -2 2a2 2 0 0 0 -2 -2a2 2 0 0 0 2 -2"
                />
                <path
                  d="M19 11h2m-1 -1v2"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="mantine-Text-root mantine-1pbxw0k"
      >
        Sidebar Text
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
